### ISSUE TITLE
[False-Positive] isset() usage - $variable !== null construction should be used when variable might not be declared

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1063]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                          |
| Plugin         | Php Inspections (EA Extended), 3.0.6.1 |
| Language level | PHP 7.2                        |

Current behaviour:
---

Here is some example code:

```php
function foo(): string
{
    $condition = false;
    if ($condition) {
        $variable = 'something';
    }

    if (isset($variable)) {
        return 'Variable is set!';
    }
    return 'Variable is not set!';
}
```

Well, obviously the code above does not follow best-practices - `$variable` is only defined if the condition is fulfilled (which in the example code is never). But the problem is, the "isset() usage" inspection suggests "'$variable !== null' construction should be used instead." and thus to replace the if check by

```php
if ($variable !== null) { ... }
```

But obviously this would lead to a notice, as `$variable` is not defined.

Expected behaviour:
---

Do not recommend to replace the above check by checking for null - the variable might not even be set.

**Comment by kalessil** [comment]
No worries, it good to re-visit the inspection from time to time =)
